mrjob/spark/mrjob_spark_harness.py

```diff
@@ -263,6 +280,7 @@ def combiner_helper(pairs1, pairs2):
         createCombiner=lambda k_v: [k_v],
         mergeValue=lambda k_v_list, k_v: combiner_helper(k_v_list, [k_v]),
         mergeCombiners=combiner_helper,
+        numPartitions=num_reducers
```

Comment:
expecting to see `preservesPartitioning=bool(num_reducers)` in the call to `mapValues()` below and inside `_discard_key_and_flatten_values()`

Reply:
http://spark.apache.org/docs/latest/api/python/pyspark.html?highlight=mapvalues#pyspark.RDD.mapValues

mapValues retains the original RDD’s partitioning therefore we don't need to (and also unable to) specify preservesPartitioning